src/System.CommandLine/StringExtensions.cs

```diff
@@ -192,10 +167,59 @@ internal static string RemovePrefix(this string option)
                 }
             }
 
-            return new LexResult(tokenList, errorList);
+            return new TokenizeResult(tokenList, errorList);
+
+            void ReadResponseFile(string arg, int i)
+            {
+                var filePath = arg.Substring(1);
+
+                if (string.IsNullOrWhiteSpace(filePath))
+                {
+                    errorList.Add(
+                        new TokenizeError(
+                            $"Invalid response file token: {arg}"));
+                    return;
+                }
+
+                try
+                {
+                    var next = i + 1;
+
+                    foreach (var newArg in ExpandResponseFile(
+                        filePath,
+                        configuration.ResponseFileHandling))
+                    {
+                        argList.Insert(next, newArg);
+                        next += 1;
+                    }
+                }
+                catch (FileNotFoundException)
+                {
+                    var message = configuration.ValidationMessages
+                                               .ResponseFileNotFound(filePath);
+
+                    errorList.Add(
+                        new TokenizeError(message));
+                }
+                catch (IOException e)
+                {
+                    var message = configuration.ValidationMessages
+                                               .ErrorReadingResponseFile(filePath, e);
+
+                    errorList.Add(
+                        new TokenizeError(message));
+                }
+            }
+        }
+
+        private static bool IsResponseFileReference(this string arg)
```

Comment:
I like this but it then means that the code using it has to understand the check and properly substring the `arg`. Perhaps change it up to something like
```suggestion
        private static string GetResponseFileReference(this string arg)
```
and simply return null if the arg is not a response file?